linera-base/src/prometheus_util.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines util functions for interacting with Prometheus (logging metrics, etc)
+
+use prometheus::{
+    register_histogram_vec, register_int_counter_vec, HistogramOpts, HistogramVec, IntCounterVec,
+    Opts,
+};
+
+const LINERA_NAMESPACE: &str = "linera";
+
+/// Wrapper arount prometheus register_int_counter_vec! macro which also sets the linera namespace
+pub fn register_int_counter_vec(
+    name: &str,
+    description: &str,
+    label_names: &[&str],
+) -> IntCounterVec {
+    let counter_opts = Opts::new(name, description).namespace(LINERA_NAMESPACE); // Setting the namespace
+    register_int_counter_vec!(counter_opts, label_names)
+        .expect("Int Counter Vec creation should not fail")
```

Comment:
I'm not sure about this. In production code one of the great things about `Result`s say versus thrown exceptions is that things *do* in fact go wrong.

There are some cases when if this happens you do want to kill the program (if you can, shut it down graciously).
There are cases where if something fails, but the caller is operationally critical you want to continue anyway.  

I'm kind of just rambling, but long story short I think this should be an error. If the caller fails - it should `error` log.

Reply:
I'm not sure I get it 🤔 won't `.expect()` still error log, but with the message we passed as an argument? If I remove it, this function will have to return a `Result`, and we'll have to "unwrap" it somehow later on anyways to use it

Reply:
I would say `.expect` is ok if the call is (provably) early at server startup but otherwise not ok, but this function cannot know, so let's have the caller decide how they want to handle the error.

These are the failure cases btw:
https://docs.rs/prometheus/latest/src/prometheus/registry.rs.html#40

Reply:
Ah I see, sounds good